packages/grid/src/GridRenderer.ts

```diff
@@ -3053,6 +3050,155 @@ export class GridRenderer {
 
     context.translate(-barLeft, -barTop);
   }
+
+  /**
+   * Gets the token boxes that are visible in the cell
+   * @param column The visible column
+   * @param row The visible row
+   * @param state The GridRenderState
+   * @returns An array of TokenBox of visible tokens or empty array
+   */
+  getTokenBoxesForVisibleCell(
+    column: VisibleIndex,
+    row: VisibleIndex,
+    state: GridRenderState
+  ): TokenBox[] {
+    return this.getCachedTokenBoxesForVisibleCell(column, row, state);
+  }
+
+  getCachedTokenBoxesForVisibleCell = memoizeClear(
+    (
+      column: VisibleIndex,
+      row: VisibleIndex,
+      state: GridRenderState
+    ): TokenBox[] => {
+      const { metrics, context, model, theme } = state;
+      const { modelRows, modelColumns } = metrics;
+
+      if (context == null || metrics == null) {
+        return [];
+      }
+
+      const modelRow = getOrThrow(modelRows, row);
+      const modelColumn = getOrThrow(modelColumns, column);
+
+      const text = model.textForCell(modelColumn, modelRow);
+      const {
+        width: textWidth,
+        x: textX,
+        y: textY,
+      } = this.getTextRenderMetrics(state, column, row);
+      const {
+        fontWidths,
+        gridX,
+        gridY,
+        width: gridWidth,
+        height: gridHeight,
+        verticalBarWidth,
+        horizontalBarHeight,
+      } = metrics;
+
+      // Set the font and change it back after
+      context.save();
+      context.font = theme.font;
+
+      const fontWidth =
+        fontWidths.get(context.font) ?? GridRenderer.DEFAULT_FONT_WIDTH;
+      const truncationChar = model.truncationCharForCell(modelColumn, modelRow);
+      const truncatedText = this.getCachedTruncatedString(
+        context,
+        text,
+        textWidth,
+        fontWidth,
+        truncationChar
+      );
+
+      const tokens = model.tokensForCell(
+        modelColumn,
+        modelRow,
+        truncatedText.length
+      );
+
+      // Check if the truncated text contains a link
+      if (tokens.length === 0) {
+        context.restore();
+        return [];
+      }
+
+      const {
+        actualBoundingBoxAscent,
+        actualBoundingBoxDescent,
+      } = context.measureText(truncatedText);
+      const linkHeight = actualBoundingBoxAscent + actualBoundingBoxDescent;
+
+      // Consider edge cases
+      const top = Math.max(gridY, gridY + textY - actualBoundingBoxAscent);
```

Comment:
The calculations for `top` and `bottom` here seem out of alignment. The bottom part of the link isn't clickable, and the top part extends past the top of the link. I suggest adding some debug code to draw the boxes as well so you can debug easier, and perhaps logging breakpoints here while testing.
Also we have a bit of a discrepancy between `getTextRenderMetrics` and `getTokenBoxesForVisibleCell`; specifically, `getTextRenderMetrics` provides the coordinates in relation to `gridX`/`gridY`, whereas `getTokenBoxesForVisibleCell` translates them by `gridX`/`gridY`. I think it's best to do the same as `getTextRenderMetrics` (e.g. not translate by `gridX`/`gridY`, and whoever calls it must do that translation).